app/form_models/offender_sar_case_form.rb

```diff
@@ -72,14 +75,40 @@ def valid?
   end
 
   def valid_attributes?(params)
-    params ||= {}
+    params ||= ActionController::Parameters.new({}).permit!
+    params = params_for_step(params, current_step)
     @case.valid_attributes?(params)
   end
 
+  def subject_type
+    object.subject_type
+  end
+
   private
 
+  def params_for_step(params, step) # rubocop:disable Metrics/CyclomaticComplexity
+    case step
+    when "subject-details"
+      params.merge!("subject_type" => "") unless params["subject_type"].present?
+      params.merge!("date_of_birth" => "") unless params["date_of_birth"].present?
+      params.merge!("flag_for_disclosure_specialists" => "") unless params["flag_for_disclosure_specialists"].present?
+    when "requester-details"
+      params.merge!("third_party": "") unless params["third_party"].present?
+      params.delete("name") unless params["third_party"] == "true"
+      params.delete("third_party_relationship") unless params["third_party"] == "true"
+
+      params.merge!("reply_method": "") unless params["reply_method"].present?
+      params.delete("email") unless params["reply_method"] == "send_by_email"
+      params.delete("postal_address") unless params["reply_method"] == "send_by_post"
+    when "date-received"
+      params.merge!("received_date" => "") unless params["received_date"].present?
+    end
+
+    params
```

Comment:
Is there a cleaner way of doing this by having say `defaults` and merging that with params if possible?

Reply:
Thanks for the suggestion! I agree it was really hard to read previously - have had a go at using methods to clarify what's going on and added a comment, is this any better?